src/ingestion/kafka-queue.ts

```diff
@@ -88,9 +91,7 @@ export class KafkaQueue implements Queue {
             await this.consumer.subscribe({ topic: KAFKA_EVENTS_INGESTION_HANDOFF })
             // KafkaJS batching: https://kafka.js.org/docs/consuming#a-name-each-batch-a-eachbatch
             await this.consumer.run({
-                // TODO: eachBatchAutoResolve: false, // don't autoresolve whole batch in case we exit it early
-                // The issue is right now we'd miss some messages and not resolve them as processEventBatch COMPETELY
-                // discards some events, leaving us with no kafka_offset to resolve when in fact it should be resolved.
```

Comment:
This comment was a bit more cryptic than it should have been, but the concern was with potentially resolving offsets for unsaved events in two cases:

* if we return early because of the `isRunning` check - however now my understanding is that if the consumer is not running, then it won't be possible to commit offsets, not completely certain that though
* if we return early because of the `isStale` check - this may happen if `consumer.seek()` or something like that is used to move partition offset somewhere, this is a bit more concerning, though also I'm unsure about the specifics of such a situation

In the end though, just to control this behavior without any doubt (we should always prefer at-least-once over at-most-once), I'd rather uncomment this (seting `eachBatchAutoResolve` to `false`) and instead just resolve the offset deliberately.
Here's the relevant comment from KafkaJS source:

<img width="765" alt="kafkajs/src/consumer/runner.js:240" src="https://user-images.githubusercontent.com/4550621/106822551-a4ff3300-667f-11eb-8fdb-6d0ab2caa098.png">

Reply:
Done